segmentation.py

```diff
@@ -0,0 +1,575 @@
+        # matrix.
+        if not (np.ndim(sigma) == 2 and
+                np.shape(sigma) == (n, n) and
+                not np.isnan(sigma).any() and
+                np.isfinite(sigma).all() and
+                np.allclose(np.transpose(sigma), sigma) and
+                linalg.det(sigma) > 0.0):
+            msg = 'The noise parameter must be a symmetric, positive-definite'
+            msg += ' matrix.'
+            raise Exception(msg)
+
+        # Check that the shape parameter is a number greater than one minus the
+        # number of degrees of freedom.
+        if not (np.isscalar(eta) and
+                not np.isnan(eta) and
+                np.isfinite(eta) and eta > n - 1.0):
+            msg = 'The shape parameter must be greater than one minus the'
+            msg += ' degrees of freedom.'
+            raise Exception(msg)
+
+        # Allocate space for storing the matrix of product statistics.
+        self.__prod = np.zeros([m + n, m + n])
+
+        # Initialize the statistics with the parameters of the prior
+        # distribution.
+        x = np.dot(omega, mu)
+        self.__prod[:m, :m] = omega
+        self.__prod[:m, m:] = x
+        self.__prod[m:, :m] = x.transpose()
+        self.__prod[m:, m:] = np.dot(np.transpose(mu), x) + eta * sigma
+        self.__weight = eta
+
+    def update(self, X, Y):
+        """Update the sufficient statistics given observed data.
+
+        The sufficient statistics represent the only parameters required to
+        describe the shape of the distribution. Due to conjugacy, the
+        sufficient statistics also form the hyper-parameters of the
+        distribution. In a Bayesian context, when the sufficient statistics are
+        referred to as 'hyper-parameters', it implies the sufficient statistics
+        reflect prior information about the distribution. After updating the
+        distribution with observed data, the sufficient statistics refer to the
+        posterior distribution.
+
+        """
+
+        # (Equation 5a, b)
+        #
+        #     | XX    XY |
+        #     | YX    YY |
+        #
+        if np.ndim(X) > 1:
+            k, m = np.shape(X)
+            x = np.dot(np.transpose(X), Y)
+
+            # Update the statistics given a block of data (in the following
+            # order: XX, XY, YX, YY)
+            self.__prod[:m, :m] += np.dot(np.transpose(X), X)
+            self.__prod[:m, m:] += x
+            self.__prod[m:, :m] += x.transpose()
+            self.__prod[m:, m:] += np.dot(np.transpose(Y), Y)
+            self.__weight += k
+
+        else:
+            m = np.size(X)
+            x = np.outer(X, Y)
+
+            # Update the statistics given a single datum.
+            self.__prod[:m, :m] += np.outer(X, X)
+            self.__prod[:m, m:] += x
+            self.__prod[m:, :m] += x.transpose()
+            self.__prod[m:, m:] += np.outer(Y, Y)
+            self.__weight += 1
+
+    def logconst(self):
```

Comment:
Given the standards-complying mood, should we change the name of this method to `getLogConstant`? This way we get to be compatible with PEP8 *and* with the [Google style guidelines](http://google-styleguide.googlecode.com/svn/trunk/cppguide.html).